wandb/data_types.py

```diff
@@ -1447,15 +1447,30 @@ def to_json(self, run_or_artifact):
                     "classes must be passed to wandb.Image which have masks or bounding boxes when adding to artifacts"
                 )
 
+            # Checks if the concrete image has already been added to this artifact
             name = artifact.get_added_local_path_name(self._path)
             if name is None:
                 name = os.path.join(
                     self.get_media_subdir(), os.path.basename(self._path)
                 )
+
+                # if not, check to see if there is a source artifact for this object
                 if (
                     self.artifact_source is not None
                     and self.artifact_source["artifact"] != artifact
                 ):
+                    # if there is, get the name of the entry (this might make sense to move to a helper off artifact)
+                    if self._path.startswith(
+                        self.artifact_source["artifact"]._default_root()
+                    ):
+                        name = self._path[
+                            len(self.artifact_source["artifact"]._default_root()) :
+                        ]
+                        # strip off the leading separator if needed
+                        if name[0] == os.sep:
+                            name = name[1:]
```

Comment:
name.lstrip(os.sep) might be more idiomatic

Reply:
incoming